### PR TITLE
Fixed CORS on regular requests. Added OPTIONS endpoint support

### PR DESCRIPTION
**Pull request by hitecSmartHome**
- Added `HTTP_OPTIONS` to the `supported_methods` list.
- Send CORS headers on every response. It is required on not just the preflight requests but on every request. ( does not fix SSE CORS problem )
- Fixed HTTPS server error. Changed it to warning if not enabled.

### CODE REVIEW

**Comment by mathieucarbou** [comment]
Thanks !
I think there are probably a few things like that to still fix in v2.

File: src/PsychicMiddlewares.h [hunk 1]
```diff
@@ -141,12 +141,12 @@ class CorsMiddleware : public PsychicMiddleware
 
     esp_err_t run(PsychicRequest* request, PsychicResponse* response, PsychicMiddlewareCallback next) override
     {
+      response->addHeader("Access-Control-Allow-Origin", _origin.c_str());
```
[the remark this answers is not in the document]
Comment (hitecSmartHome) [reply src/PsychicMiddlewares.h:144]: I have tested this in brave and chrome and the requests are failing if we don't send them even on non cors requests.

[the remark this answers is not in the document]
Comment (hitecSmartHome) [reply src/PsychicMiddlewares.h:144]: Will paste in some proof

**Review by mathieucarbou — COMMENTED** [review]
just a comment

File: src/PsychicMiddlewares.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (mathieucarbou) [reply src/PsychicMiddlewares.h:144]: > I have tested this in brave and chrome and the requests are failing if we don't send them even on non cors requests.

what don you mean by non-cors ?

**Pull request merged**